### PR TITLE
License work

### PR DESCRIPTION
**Pull request by joshfischer1108**
@nwangtw This is an attempt to address the licensing issues brought up in our last release.  
The email thread for the last release is [here](https://mail-archives.apache.org/mod_mbox/incubator-general/201907.mbox/%3CCAFkuAo0heb65W6%2BX%3DNgmLxK0XY4WPK0cuM5EWxWW%2B%3DHDhNrotA%40mail.gmail.com%3E).  This is my first attempt to work on the licenses in the project.  Hopefully it is not too far off and will get us a step closer to releasing binaries for the project.

For convenience I picked through to copy and paste comments from the last release thread on general@incubator that addressed issues with the licenses in the project below this line

```
The third party licenses could use a little work. The License text should be included in LICENSE or pulled into the licenses directory. Any required copyright attribution should be in the NOTICE. Please create an issue to fix these for the next release. 
RAT Check. The *.md files should have Apache License Headers and CONTRIBUTING.md is link where the target is missing in the release.
 Please create an issue to fix .rat-excludes and fix the md files license. Please create an issue to annotate .rat-excludes to explain any that are excludes related to other licensing.
```
```
NOTICE has incorrect year (please update for next time)
LICENSE is missing a number of permissive licenses (I didn't check ones
Justin had already covered but third_party\python\semver seems to be
missing from list of BSD licensed software though it does appear in your
rat excludes - please add for next release)
```
```
Licence is misisng
-  MIT licensed software copyright  Facebook [1][2][3][4][5][6]
- ALv2 files [7][8]
- ALv2 files [9][10][11][12][13][14][15][16]
- this ALv2 file [17]
- this  ALv2 file [18]
- this ALv2 file [19]
- this file [20] includes Sizzle (MIT, BSD, and GPL Licenses)
- this file [21] includes modernizr (MIT licensed)

(I just did a search for the word copyright and match what was listed in the LICENSE file if you’re wondering how I found these)

1. ./website2/website/core/Footer.js
2. ./website2/website/siteConfig.js
3. ./website2/website/pages/en/users.js
4. ./website2/website/pages/en/index.js
5. ./website2/website/pages/en/help.js
6. ./website2/website/pages/en/versions.js
7. ./tools/rules/pex/pex_rules.bzl
8. ./tools/rules/pex/wrapper/pex_wrapper.py
9. ./scripts/setup-eclipse.sh
10. ./scripts/detect_os_type.sh
11. ./scripts/get_all_heron_paths.sh
12. ./scripts/packages/template_bin.sh
13. ./scripts/packages/self_extract_binary.bzl
14. ./scripts/packages/package_info_generator.sh
15. ./scripts/packages/tests_template_bin.sh
16. ./scripts/packages/bin_common.sh
17. ../heron/shell/src/python/handlers/healthhandler.py
18. ./third_party/java/bazel/extra_actions_base.proto
19. ./third_party/java/jarjar/src/main/resources/com/tonicsystems/jarjar/help.txt
20. ./heron/shell/assets/jquery.js
21 ./heron/tools/ui/resources/static/js/bootstrap.js

```

### CODE REVIEW

**Comment by nwangtw** [comment]
Thanks a lot!

Licenses are tricky. We are not sure about all the works, but this is one step forward!

**Comment by jemantheiy** [comment]
Looks good so far.

File: LICENSE [hunk 1]
```diff
@@ -292,3 +323,13 @@ See project link for details.
 
     kashmir(https://github.com/Corvusoft/kashmir-dependency/blob/master/LICENSE_1_0.txt)
       -> third_party/kashmir/*
+
+========================================================================
+Third party GNU General Public License, Version 3.0 licenses
```
[the remark this answers is not in the document]
Comment (joshfischer1108) [reply LICENSE]: @nwangtw  I believe you are correct.  Who would be the person to remove this library?  I'm not familiar with where or how it is used in the Heron repo.

[the remark this answers is not in the document]
Comment (nwangtw) [reply LICENSE]: We will need to figure out how it is used.

[the remark this answers is not in the document]
Comment (dave2wave) [reply LICENSE]: Look at the source file referenced by the GPL license inclusion.

/*!
 * jQuery JavaScript Library v1.7.2
 * http://jquery.com/
 *
 * Copyright 2011, John Resig
 * Dual licensed under the MIT or GPL Version 2 licenses.
 * http://jquery.org/license
 *
 * Includes Sizzle.js
 * http://sizzlejs.com/
 * Copyright 2011, The Dojo Foundation
 * Released under the MIT, BSD, and GPL Licenses.
 *
 * Date: Wed Mar 21 12:46:34 2012 -0700
 */

When a file is licensed under multiples we choose the one that is compatible. In these cases we do NOT choose GPL.

[the remark this answers is not in the document]
Comment (joshfischer1108) [reply LICENSE]: Fantastic.  Thank you for the guidance @dave2wave

**Pull request merged**